### PR TITLE
feat: implemented activate subscription

### LINKED ISSUE #161: Implement Activate Subscription Endpoint
Description:
We need to create an endpoint that activates a subscription. This endpoint will allow users or organisation members to activate their own subscription.

Purpose:
The purpose of this endpoint is to provide a way to activate a subscription, changing its status from inactive to active and potentially updating other related fields.

Expected Outcome:
A successful call to this endpoint should activate the specified subscription, updating its status in the database and returning the updated subscription details.

Requirements:
1. Integrate with our database system
2. Implement authorization checks
3. Error handling and logging
4. CQRS pattern implementation
5. Use of generic response DTOs

Acceptance Criteria:
1. The endpoint successfully activates the subscription in the database
2. It returns the expected response format on success
3. It handles and returns appropriate error responses
4. The endpoint is secured with proper authentication and authorization
5. Users or organisation members can only activate their own subscription
6. Admins and superadmins can activate any subscription
7. The endpoint follows CQRS pattern and uses generic response DTOs
8. The activation process handles different subscription plans correctly

Endpoint Details:
- URL: /api/v1/subscriptions/{subscriptionId}/activate
- Method: POST
- Authentication: Required (Bearer token)
- Authorization: User/Organisation Member (own subscription only), Admin, SuperAdmin

Request Body:
{
    "subscriptionId": "subscription-guid-here"
}

Successful Response Body (200 OK):
{
  "data": {
    "id": "subscription-guid",
    "plan": "Basic",
    "frequency": "Monthly",
    "isActive": true,
    "startDate": "2024-07-27T00:00:00Z",
    "amount": 1000,
    "userId": "user-guid-here",
    "organisationId": "org-guid-here",
    "email": "user@example.com"
  },
  "message": "Subscription activated successfully"
}

Error Response (400 Bad Request):
{
  "error": "Invalid subscription ID. Please provide a valid subscription ID",
  "message": "Request failed"
}

Error Response (404 Subscription Not Found):
{
  "error": "Subscription not found. Please check the subscription ID and try again."
  "message": "Request failed"
}

Error Handling:
* Invalid subscription ID (400 Bad Request)
* Authentication failure (401 Unauthorized)
* Authorization failure (403 Forbidden)
* Subscription not found (404 Not Found)
* Subscription already active (409 Conflict)
* Internal server errors (500 Internal Server Error)

Security Measures:
1. Use HTTPS for all API communications
2. Implement rate limiting to prevent abuse
3. Validate and sanitize all input parameters
4. Use environment variables for storing sensitive data
5. Implement proper authentication and authorization checks

Tasks:
1. Create Activate Subscription Command and handler
2. Implement controller action for the endpoint
3. Implement authorization checks
4. Implement logic for handling different subscription plans
5. Implement error handling and logging
6. Write unit and integration tests
7. Add Swagger documentation for the API endpoint
8. Perform security review and penetration testing
9. Deploy to staging environment for testing

### PR DESCRIPTION
**Pull request by AlakaTahir**
# Description

Closes #161 **

# Changes proposed

## What did you do?

Develop endpoint to activate a subscription, changing its status from inactive to active and potentially updating other related fields.

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [ ] My code follows the code style of this project.
- [ ] This PR does not contain plagiarized content.
- [ ] The title and description of the PR is clear and explains the approach.
- [ ] I am making a pull request against the **dev branch** (left side).
- [ ] My commit messages styles matches our requested structure.
- [ ] My code additions will fail neither code linting checks nor unit test.
- [ ] I am only making changes to files I was requested to.

# Screenshots/Videos

### CODE REVIEW

File: src/Hng.Application/Features/Subscriptions/Dtos/SubscriptionDto.cs
Comment (Ace2489) [inline src/Hng.Application/Features/Subscriptions/Dtos/SubscriptionDto.cs:1]: Snake case for the property names, please

**Comment by Ace2489** [comment]
Just a couple of fixes and you should be good to go.

File: src/Hng.Application/Features/Subscriptions/Handlers/ActivateSubscriptionCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+public class ActivateSubscriptionCommandHandler : IRequestHandler<ActivateSubscriptionCommand, SubscriptionDto>
+{
+    private IRepository<Subscription> _subscriptionRepository;
+    private IMapper _mapper;
+
+
+    public ActivateSubscriptionCommandHandler(IRepository<Subscription> subscriptionRepository, IMapper mapper)
+    {
+		_subscriptionRepository = subscriptionRepository;
```
Comment (chimaihueze) [inline src/Hng.Application/Features/Subscriptions/Handlers/ActivateSubscriptionCommandHandler.cs:18]: Too much indentation. Please fix this.

File: src/Hng.Application/Features/Subscriptions/Handlers/ActivateSubscriptionCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    public async Task<SubscriptionDto> Handle(ActivateSubscriptionCommand request, CancellationToken cancellationToken)
+    {
+        var data = await _subscriptionRepository.GetBySpec(
+			o => o.Id == request.SubscriptionId);
+
+        if(data != null)
+        {
+			data.IsActive = true;
+			data.UpdatedAt = DateTime.UtcNow;
```
Comment (chimaihueze) [inline src/Hng.Application/Features/Subscriptions/Handlers/ActivateSubscriptionCommandHandler.cs:30]: Too much indentation. Please fix.

**Review by chimaihueze — CHANGES_REQUESTED** [review]

**Review by Godhanded — CHANGES_REQUESTED** [review]
In your project root directory run
dotnet format --no-restore

to fix all formatting issues automatically

**Review by Godhanded — CHANGES_REQUESTED** [review]
Okay sweet, please add write ups to the PR template headings are there but no details about what this PR is and what issue it closes, feel free to create the issue if it doesn't already exist on the GitHub.

**Review by Godhanded — APPROVED** [review]

**Pull request merged**